### ISSUE TITLE
environment variable

### ISSUE THREAD

**Issue by maxbachmann** [piwheels/piwheels#319]
in my package I try to build a C++ version of the package, but if this fails I build a pure Python version. This version is slower, but at least works. Both Cibuildwheel and conda-forge provide an environment variable to indicate they are building the package. This allows me to disable this behaviour, since when building the package you really want the C++ version to be built. Does piwheels set a similar flag?

**Comment by bennuttall** [comment]
No, but we can try to resolve the building issue if you open an issue on our [package issue tracker](https://github.com/piwheels/packages/issues) via the links at the bottom of the project page e.g. https://www.piwheels.org/project/numpy/

**Comment by maxbachmann** [comment]
@bennuttall the package builds fine. The setup.py of the package is essentially:
```python
if packaging:
    build_binary()
else:
    try:
        build_binary()
    except:
        build_pure_python()
```
The only problem is that in case the build is ever broken, it will just build a wheel for the pure Python version. This is something you really do not want for binaries in a package manager.  For `conda` and `cibuildwheel` I ensure that they never build pure python wheels using their corresponding environment variables.
```
packaging = "1" in {
    os.environ.get("CIBUILDWHEEL", "0"),
    os.environ.get("CONDA_BUILD", "0"),
    os.environ.get("RAPIDFUZZ_BUILD_EXTENSION", "0")
}
```

It would make sense to do the same for piwheels. However this requires a way to detect that the piwheels packaging.

**Comment by bennuttall** [comment]
Yes we can consider doing that in future if it helps. @waveform80 

But surely if it fails at the `build_binary` stage anyway, and that's something that will be present in the logs and something we can fix. We provide access to our build logs on project pages so you can see them too. If you open an issue for your package we can take a look.

**Comment by maxbachmann** [comment]
Yes, it would be visible in the build logs. However I only maintain the PyPi and conda packages myself. For others like piwheels, I largely rely on others to report issues with failed builds. It's much easier for users to spot a failed build than to check the log messages of a successful build.

**Comment by bennuttall** [comment]
What's the name of the package?

**Comment by maxbachmann** [comment]
rapidfuzz: https://github.com/maxbachmann/RapidFuzz

**Comment by bennuttall** [comment]
I've just rebuilt rapidfuzz and it's successfully built a platform wheel (rather than a pure Python wheel). Is that what you wanted? https://www.piwheels.org/project/rapidfuzz/ The build log for 2.1.1 on cp39 is at https://www.piwheels.org/logs/0000/0848/3546.txt

**Comment by bennuttall** [comment]
The same applies for jellyfish: https://www.piwheels.org/project/jellyfish/

**Comment by maxbachmann** [comment]
> I've just rebuilt rapidfuzz and it's successfully built a platform wheel (rather than a pure Python wheel).

I was actually not even aware that it did not build those wheels for the latest version. What I mean is that in case the compilation fails there will only be a warning in the build log:
```
==========================================================================
WARNING: The C extension could not be compiled, speedups are not enabled.
Failure information, if any, is above.
Retrying the build without the C extension now.
==========================================================================

==========================================================================
WARNING: The C extension could not be compiled, speedups are not enabled.
Plain-Python build succeeded.
==========================================================================
```
this behavior makes sense when installing directly from the sdist, since it is still better to have a slower version of the lib than none. However in the case of piwheels it would mean that in case of a failing build a pure Python version of the library placed inside the binary wheel. In conda-forge / cibuildwheels builds I prevent this from happening by disabling the fallback behavior.

**Comment by bennuttall** [comment]
> I was actually not even aware that it did not build those wheels for the latest version

The builds had failed because pip found no matching distribution at the time. I think this must be when PyPI recently had issues with files not syncing quickly enough so when we thought the version was available, there was nothing found. But trying again now meant it found sdist and built fine.

> What I mean is that in case the compilation fails there will only be a warning in the build log:

I understand. We'll aim to introduce an environment variable you can use in future. For now I just want to check you're happy with the current built files.

**Comment by maxbachmann** [comment]
> I understand. We'll aim to introduce an environment variable you can use in future.

:+1:

> For now I just want to check you're happy with the current built files.

yes those builds look fine

**Comment by maxbachmann** [comment]
@bennuttall this feature would be super useful. I have broken builds again and only realized it by luck a month later, since the builds still pass.
